pkg/kubeapiserver/options/authentication.go

```diff
@@ -566,3 +586,30 @@ func (o *BuiltInAuthenticationOptions) ApplyAuthorization(authorization *BuiltIn
 		o.Anonymous.Allow = false
 	}
 }
+
+func (o *BuiltInAuthenticationOptions) validateOIDCOptions() []error {
+	var allErrors []error
+
+	// Existing validation when jwt authenticator is configured with oidc-* flags
+	if len(o.AuthenticationConfigFile) == 0 {
+		if o.OIDC != nil && (len(o.OIDC.IssuerURL) > 0) != (len(o.OIDC.ClientID) > 0) {
+			allErrors = append(allErrors, fmt.Errorf("oidc-issuer-url and oidc-client-id should be specified together"))
+		}
+
+		return allErrors
+	}
+
+	// New validation when authentication config file is provided
+
+	// Authentication config file is only supported when the StructuredAuthenticationConfiguration feature is enabled
+	if !utilfeature.DefaultFeatureGate.Enabled(genericfeatures.StructuredAuthenticationConfiguration) {
+		allErrors = append(allErrors, fmt.Errorf("Set --feature-gates=%s=true to use authentication-config file", genericfeatures.StructuredAuthenticationConfiguration))
+	}
+
+	// Authentication config file and oidc-* flags are mutually exclusive
+	if o.OIDC != nil && (len(o.OIDC.IssuerURL) > 0 || len(o.OIDC.ClientID) > 0) {
+		allErrors = append(allErrors, fmt.Errorf("authentication-config file and oidc-* flags are mutually exclusive"))
+	}
```

Comment:
Haven't updated this check to verify every single `oidc-*` flag yet. The tricky one will be `--oidc-username-claim` (default value _sub_) and `--oidc-signing-algs` (default value _[]string{"RS256"}_). The non-default value check will be easy but it's possible the user configured the flags explicitly with the default value. Given the authenticator setup will fail just without these 2 flags, I'm wondering if we should check all the flags? LMK what you think!

Reply:
Don't flags have the ability to distinguish between being explicitly set vs just defaulted?

Reply:
IIRC, there is an option to check if the flag has changed. I'll look into it.

Reply:
Updated to check using [FlagSet.Changed](https://pkg.go.dev/github.com/spf13/pflag@v1.0.5#FlagSet.Changed).